server/src/watchdog.js

```diff
@@ -138,3 +149,45 @@ exports.submit = function(shot) {
     mozlog.error("watchdog-failed-submission", {err: e});
   });
 };
+
+exports.handleResult = function(req) {
+  getSubmission(req.params.submissionId).then(record => {
+    if (record.request_id !== req.body.watchdog_id) {
+      throw new Error(`Received mismatching Watchdog ID in callback for submission ${record.id}.`);
```

Comment:
Sentry doesn't do well with dynamically-generated error messages.

This pattern works in the add-on, but I'm unsure if it works on the server:

```js
let exc = new Error("Received mismatching Watchdog ID in callback");
exc.recordId = String(record.id); // Note, if you don't turn it into a string then "undefined" gets lost
exc.watchDogId = String(req.body.watchdog_id);
```

But... I'm really not sure if that works.